app/models/ext_management_system.rb

```diff
@@ -712,6 +740,19 @@ def self.display_name(number = 1)
     n_('Manager', 'Managers', number)
   end
 
+  protected
+
+  # Child managers went to/from maintenance mode with parent
+  def change_maintenance_for_child_managers
+    child_managers.each do |child_ems|
+      if enabled?
+        child_ems.resume!
+      else
+        child_ems.pause!(backup_zone)
+      end
```

Comment:
This feels like it could be encapsulated in a method on the EMS itself, then this method would just be

```
child_managers.each(&:toggle_enabled!)
```

or something like that.  Not required, just feels like a method that should be owned by the EMS.

Reply:
by this, it implies that EMS's cannot be nested in more than 1 level never in the future for any future provider, that's why i prefer calling children
moreover - `enabled/disabled` is subset of `resume/pause` in this case